### PR TITLE
Whether or not a pacifist can fire a gun is no longer based on the gun being harmful, but the chambered round being harmful

### LINKED ISSUE #36600: Pacifists can't use less-than-lethal rounds + can't use DRAGnet
Round ID: Irrelevant

Reproduction:
Step 1) Load shotgun with beanbag rounds or stun rounds.
Step 2) Be unable to fire.

This probably isn't an easy tweak to make at all and I'm not sure if it's a feature but if it is then that is a bit stupid.

To fix this, you'd have to make the game check what ammo type is chambered every time you wanted to fire, and that could cause issues with burst fire weapons, either stopping you mid burst, or allowing players to autisticly load rounds so that they could fire a non-lethal round and then finish the burst with lethals.

However, it would be much more modular than just designating guns as "harmful" and "non-harmful"
Edit: edited for clarity.

### PR DESCRIPTION
**Pull request by ghost**
:cl:
fix: makes dragnet non harmful
tweak: pacifists can now use any disabler or stun setting on any energy gun
code: removed all of the pacifism check code from code/modules/mob/living/living.dm
code: gun objects no longer have a harmful variable, instead, ammo_casing objects now have a harmful variable, which is by default set to TRUE
code: if a pacifist fires a gun, it checks whether or not the round chambered is lethal, instead of whether or not the gun itself is lethal.
/:cl:

resolves #36600 
[video link](https://youtu.be/CkM6rn7Bz4s)

### CODE REVIEW

**Comment by Rinoka** [comment]
as mentioned in the issue, it has some... debatable implications (pacifists with a gun they didn't load able to sense the lethality of the bullets)

but if we gonna merge we gonna merge, i'm just raising the potential issue so we know what we want pacifism to be

**Comment by Rinoka** [comment]
also, wasn't sure: are pacifists able to use syringe guns

**Comment by iskyp** [comment]
no

**Comment by Rinoka** [comment]
ok good

those would be A Mess

**Comment by Iamgoofball** [comment]
Knocking people down is inherently harmful

**Comment by iskyp** [comment]
open an issue about how the trait isn't realistic enough then

**Comment by Rinoka** [comment]
having goof _still_ not be gitbanned is inherently harmful

and yet here we are

**Comment by iskyp** [comment]
FUCK BURST FIRE WEAPONS WORK HOLD ON

**Comment by chaoko99** [comment]
Pacifist, being one who pacifies. That actually doesn't explicitly deny killing but... It's better meant as subdued here.

**Comment by AlexTheSysop42** [comment]
> pacifists with a gun they didn't load able to sense the lethality of the bullets

I assume you're talking about ballistic weapons, and I don't think there are any of those that aren't harmful. (Even the detective's gun does some damage.)

**Comment by iskyp** [comment]
soporific rounds for the .50 BMG are nonlethal but I haven't designated that because they aren't like that currently

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -212,6 +211,10 @@
 			firing_burst = FALSE
 			return FALSE
 	if(chambered && chambered.BB)
+		if(user.has_trait(TRAIT_PACIFISM))
+			if(chambered.harmful)
+				to_chat(user, "<span class='notice'> [src] is lethal! You don't want to risk harming anyone...</span>")
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/gun.dm]: The gun doesn't matter, the bullet inside it does. The notification should explain this.

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -257,6 +260,10 @@
 			addtimer(CALLBACK(src, .proc/process_burst, user, target, message, params, zone_override, sprd, randomized_gun_spread, randomized_bonus_spread, rand_spr, i), fire_delay * (i - 1))
 	else
 		if(chambered)
+			if(user.has_trait(TRAIT_PACIFISM))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/gun.dm]: shooting something should not check if you have a trait, having a certain trait should pick up when you shoot something.

Then again this is guncode so...

File: code/modules/projectiles/gun.dm [hunk 3]
```diff
@@ -257,6 +260,10 @@
...
+				if(chambered.harmful)
+					to_chat(user, "<span class='notice'> [src] is lethal! You don't want to risk harming anyone...</span>")
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/projectiles/gun.dm]: > The gun doesn't matter, the bullet inside it does. The notification should explain this.

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
also you want `fixes #36600` in the PR body.

I still disagree pacifism should be able to stun at all in a stun-combat system we have currently, and believe the actual fix would be to remove the ability to do such rather than provide more outlets to subdue individuals.

**Comment by AlexTheSysop42** [comment]
The whole thing about a stun-combat system is that once someone's stunned, the attacker has to actually hit the them to kill them, otherwise they just get back up. Pacifists can't use the stun to kill.

**Comment by iskyp** [comment]
@ExcessiveUseOfCobblestone 
> i was slightly upset and made #36600 then decided to fix it
also added commenting to appease you

**Comment by ExcessiveUseOfCobblestone** [comment]
I wasn't saying it's unclear, I'm saying that's not the correct way to approach it if possible.

Checking if the gun is harmful when you try to shoot it should be on pacifism rather than checking if the individual has pacifism everytime you shoot the gun.

**Comment by iskyp** [comment]
ill see what I can do when I get back home

**Comment by Anonmare** [comment]
Going by the original pacifists, The Ai and its metal legion, tasers and disablers are not harmful so this would be fine albeit a technicality

**Comment by Iamgoofball** [comment]
AIs aren't pacifist buddy they just don't want harm occurring to humans, they'll go full murderbot on non humans

**Comment by Rinoka** [comment]
They still are technically the _original_ pacifists, considering all that complex nonhuman bullshit is just meta rearing its head and back in the day AIs would still consider antags human, which resulted in them requiring pacifism.

You're not wrong nowadays that we get full murderbot AIs but still

**Comment by iskyp** [comment]
so after dicking around with code for a few days i don't know how to tie the code directly to the trait like @ExcessiveUseOfCobblestone suggested
I've gotten it to work somewhat
1) change `if(shooty && !shooty.harmful)` on line 860 of living.dm to 
`if(shooty && !shooty.chambered.harmful)`
medbeam is rendered unable to fire because it lacks an ammo_casing and breaks
2) the code i currently have
works perfectly but might be less than ideal in terms of code optimization

if someone could nudge me onto the right path if there is one that'd be pretty cool because i'm still fairly new to dm

**Comment by ExcessiveUseOfCobblestone** [comment]
Don't worry i'll do it later

**Comment by iskyp** [comment]
yes, pacifists can cuff

**Comment by Bawhoppen** [comment]
No better way to prevent violence, than to lock everybody up.

**Pull request merged**